datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx

```diff
@@ -0,0 +1,327 @@
+            result.entity.type,
+        );
+
+        const tag = renderTag(
+            displayName,
+            (result.entity as Tag).urn,
+            (result.entity as Tag).properties?.colorHex,
+            result.entity.type,
+        );
+        return result.entity.type === EntityType.Tag ? (
+            <Select.Option value={`${tag.value}${NAME_TYPE_SEPARATOR}${tag.type}`} key={tag.value}>
+                {tag.label}
+            </Select.Option>
+        ) : (
+            <Select.Option value={`${term.value}${NAME_TYPE_SEPARATOR}${term.type}`} key={term.value}>
+                {term.label}
+            </Select.Option>
+        );
+    };
+
+    const tagSearchOptions = tagSearchResults.map((result) => {
+        return renderSearchResult(result);
+    });
+
+    const inputExistsInTagSearch = tagSearchResults.some((result: SearchResult) => {
+        const displayName = entityRegistry.getDisplayName(result.entity.type, result.entity);
+        return displayName.toLowerCase() === inputValue.toLowerCase();
+    });
+
+    if (!inputExistsInTagSearch && inputValue.length > 0 && type === EntityType.Tag && urnIds.length === 0) {
+        tagSearchOptions.push(
+            <Select.Option value={CREATE_TAG_VALUE} key={CREATE_TAG_VALUE}>
+                <Typography.Link> Create {inputValue}</Typography.Link>
+            </Select.Option>,
+        );
+    }
+
+    const tagRender = (props) => {
+        // eslint-disable-next-line react/prop-types
+        const { label, value, closable, onClose } = props;
+        const { type: selectedType } = getSelectedValue(value);
+        const onPreventMouseDown = (event) => {
+            event.preventDefault();
+            event.stopPropagation();
+        };
+        return (
+            <CustomTag
+                onMouseDown={onPreventMouseDown}
+                closable={closable}
+                onClose={onClose}
+                style={{
+                    padding: selectedType === EntityType.Tag ? '0px 7px 0px 0px' : '0px 7px',
+                    marginRight: 3,
+                    borderRadius: '100em',
+                    lineHeight: 0,
+                    display: 'flex',
+                    justifyContent: 'start',
+                    alignItems: 'center',
+                }}
+            >
+                {label}
+            </CustomTag>
+        );
+    };
+
+    // Handle the Enter press
+    useEnterKeyListener({
+        querySelectorToExecuteClick: '#addTagButton',
+    });
+
+    if (showCreateModal) {
+        return (
+            <CreateTagModal
+                visible={visible}
+                onClose={onCloseModal}
+                onBack={() => setShowCreateModal(false)}
+                tagName={inputValue}
+                entityUrn={entityUrn}
+                entitySubresource={entitySubresource}
+            />
+        );
+    }
+
+    const getUrnId = (urnId: string) => {
```

Comment:
Let's rename all instances of "urnId" to just "urn"

An urn is implied to be an ID.